### PR TITLE
Support this new configuration approach for Reconciler image bump in control-plane job

### PR DESCRIPTION
**Pull request by ruanxin**
**Description**

Changes proposed in this pull request:

- relocate reconciler-image-autobump file from reconciler repo to test-infra
- introduce a customized autobumper source code into test-infra, disabled some features (replace image tag in target file, commit changes) of autobumper, only left the part to create PR.
- refactor `prow/scripts/reconciler-image-autobump.sh` script, introduce new process to replace image tag in target file, commit changes to `kyma-bot:control-plane` repo

**Related issue(s)**
https://github.com/kyma-incubator/reconciler/issues/1035

### CODE REVIEW

File: prow/scripts/reconciler-image-autobump.sh [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  RECONCILER_COMMIT_ID="$(git rev-parse HEAD)"
+  export RECONCILER_IMAGE_TAG="$(echo "${RECONCILER_COMMIT_ID}" | cut -c1-8)"
+  log::info "Reconciler image tag: ${RECONCILER_IMAGE_TAG}"
+}
+
+function autobump::update_reconciler_image_tag(){
+  log::info "Update reconciler image tag in control plane"
+  cd "${CONTROL_PLANE_DIR}"
+  # support old image tag update, should be removed after PR https://github.com/kyma-project/control-plane/pull/1601 merged.
+  yq e -i '(.global.images.mothership_reconciler ) |= "eu.gcr.io/kyma-project/incubator/reconciler/mothership:'${RECONCILER_IMAGE_TAG}'"' ./resources/kcp/values.yaml
```
Comment (Tomasz-Smelcerz-SAP) [inline prow/scripts/reconciler-image-autobump.sh:83]:
```suggestion
  yq e -i '(.global.images.mothership_reconciler ) = "eu.gcr.io/kyma-project/incubator/reconciler/mothership:'${RECONCILER_IMAGE_TAG}'"' ./resources/kcp/values.yaml
```

File: prow/scripts/reconciler-image-autobump.sh [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  yq e -i '(.global.images.component_reconciler ) |= "eu.gcr.io/kyma-project/incubator/reconciler/component:'${RECONCILER_IMAGE_TAG}'"' ./resources/kcp/values.yaml
```
Comment (Tomasz-Smelcerz-SAP) [inline prow/scripts/reconciler-image-autobump.sh:84]:
```suggestion
  yq e -i '(.global.images.component_reconciler ) = "eu.gcr.io/kyma-project/incubator/reconciler/component:'${RECONCILER_IMAGE_TAG}'"' ./resources/kcp/values.yaml
```

File: prow/scripts/reconciler-image-autobump.sh [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+  yq e -i '(.global.images.mothership_reconciler_version ) |= "'${RECONCILER_IMAGE_TAG}'"' ./resources/kcp/values.yaml
```
Comment (Tomasz-Smelcerz-SAP) [inline prow/scripts/reconciler-image-autobump.sh:86]:
```suggestion
  yq e -i '(.global.images.mothership_reconciler_version ) = "'${RECONCILER_IMAGE_TAG}'"' ./resources/kcp/values.yaml
```

File: prow/scripts/reconciler-image-autobump.sh [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  yq e -i '(.global.images.components.[] | select(has("version")).["version"] ) |= "'${RECONCILER_IMAGE_TAG}'"' ./resources/kcp/values.yaml
```
Comment (Tomasz-Smelcerz-SAP) [inline prow/scripts/reconciler-image-autobump.sh:87]:
```suggestion
  yq e -i '(.global.images.components.[] | select(has("version")).["version"] ) = "'${RECONCILER_IMAGE_TAG}'"' ./resources/kcp/values.yaml
```

**Review by Tomasz-Smelcerz-SAP — DISMISSED** [review]

**Review by tobiscr — APPROVED** [review]

**Review by m00g3n — APPROVED** [review]
LGTM

**Review by Tomasz-Smelcerz-SAP — APPROVED** [review]

**Review by tobiscr — APPROVED** [review]

**Review by jakobmoellersap — APPROVED** [review]

**Comment by Ressetkk** [comment]
First of all - do not introduce another go.mod. We aim to have all tooling under one go module.
Please move go code outside of the `resources` catalogue. It's not a resource, but a source code. This should lay in the `development` folder.

Why there was a need to strip the autobumper code? Can't it be used as-is? Autobumber code should be able to work without any additional shell scripts, and it would be best rather to avoid using them and have the binary work as container entrypoint.

/hold

**Comment by ruanxin** [comment]
Hi @dekiel @Ressetkk, thanks for spend the time to review and come up a temporary solution with us. Here is the followup [issue](https://github.com/kyma-incubator/reconciler/issues/1094) to remove this pipeline and related code, I also created a revisit meeting for us (and you already accepted) to verify if the code still there in code base and think about further actions.

If there is no further issue, please accept this PR.

**Review by dekiel — APPROVED** [review]

**Comment by Ressetkk** [comment]
Please don't forget about this codebase, as we agreed upon it. @ruanxin
/unhold

**Comment by ruanxin** [comment]
Meeting notes on Oct. 17:

- Revisited this issue, had agreement with Neighbors, keep this temporary solution in the code base until old reconciler get replaced with new operator solution.
- Setup followup revisit meeting after 5 month.

File: prow/scripts/reconciler-image-autobump.sh [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+set -e
+
+ENABLE_TEST_LOG_COLLECTOR=false
+
+# Exported variables
+export K8S_TEST_INFRA_SOURCES_DIR="/home/prow/go/src/github.com/kubernetes/test-infra"
+export RECONCILER_DIR="/home/prow/go/src/github.com/kyma-incubator/reconciler"
+export CONTROL_PLANE_DIR="/home/prow/go/src/github.com/kyma-project/control-plane"
+export KYMA_TEST_INFRA_SOURCES_DIR="/home/prow/go/src/github.com/kyma-project/test-infra"
+export BUMP_TOOL_CONFIG_FILE="${KYMA_TEST_INFRA_SOURCES_DIR}/prow/scripts/resources/control-plane-autobump-reconciler-config.yaml"
```
Comment (ruanxin) [inline prow/scripts/reconciler-image-autobump.sh:28]: This `BUMP_TOOL_CONFIG_FILE` need to update after code relocated to reconcile repo

File: prow/scripts/reconciler-image-autobump.sh [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+## Configurations and Variables
+## ---------------------------------------------------------------------------------------
+
+# exit on error, and raise error when variable is not set when used
+set -e
+
+ENABLE_TEST_LOG_COLLECTOR=false
+
+# Exported variables
+export K8S_TEST_INFRA_SOURCES_DIR="/home/prow/go/src/github.com/kubernetes/test-infra"
```
Comment (ruanxin) [inline prow/scripts/reconciler-image-autobump.sh:24]: `K8S_TEST_INFRA_SOURCES_DIR` can be removed after code relocated

File: templates/data/incubator-buildpack-data.yaml [hunk 7]
```diff
@@ -292,7 +292,7 @@ templates:
             annotations:
               pipeline.installer: reconciler
           reconciler_image_autobump_jobConfig:
-            command: "/home/prow/go/src/github.com/kyma-incubator/reconciler/scripts/prow/reconciler-image-autobump.sh"
+            command: "/home/prow/go/src/github.com/kyma-project/test-infra/prow/scripts/reconciler-image-autobump.sh"
```
Comment (ruanxin) [inline templates/data/incubator-buildpack-data.yaml:295]: the script location in command should be changed

**Pull request merged**